modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java

```diff
@@ -177,6 +179,31 @@ public final void collectMetrics(Request<?> request, Response<?> response) {
                 .map(List::size)
                 .orElse(0);
 
+            if (exceptionCount > 0) {
+                final List<Object> statusCodes = Objects.requireNonNullElse(
+                    awsRequestMetrics.getProperty(AWSRequestMetrics.Field.StatusCode),
+                    List.of()
+                );
+                // REQUESTED_RANGE_NOT_SATISFIED errors are expected errors due to RCO
+                // TODO Add more expected client error codes
+                final long amountOfRequestRangeNotSatisfiedErrors = statusCodes.stream()
+                    .filter(e -> (Integer) e == REQUESTED_RANGE_NOT_SATISFIED.getStatus())
+                    .count();
+                if (amountOfRequestRangeNotSatisfiedErrors > 0) {
+                    var clientErrorAttributes = Maps.copyMapWithAddedEntry(
+                        attributes,
+                        "error_code",
```

Comment:
nit I'm not sure the additional attribute for the "error_code" is necessary. Since we count the REQUESTED_RANGE_NOT_SATISFIED separately, we know they are the 416 error so there's no reason to redundantly store the error_code as an additional attribute.